### DISCUSSION TITLE
Pre/Post tasks

### DISCUSSION THREAD

**Discussion by adepretis** [copier-org/copier#967] [Q&A]
Is there a way to have pre- and post-generation tasks like e.g. Cookiecutter has with it's `hooks`. Currently it seems those Copier tasks are only post-generation tasks.

My use case:

- generate Laravel (insert Framework X) boilerplate using a pre-generation tasks/hooks, like
  - `composer create-project --prefer-dist laravel/laravel myapp` or
  - `curl -s "https://laravel.build/example-app" | bash`
- overwrite some generated files with Copier template files or inject new files into the resulting directory structure
- run post-generation tasks/hooks for cleanup

With the current `tasks` implementation I don't see a way to do this.

Any ideas?

**Comment by pawamoy** [comment]
Tasks need some love. See discussion in https://github.com/copier-org/copier/issues/240.
